packages/app/src/background/services/credentials/index.ts

```diff
@@ -199,6 +200,62 @@ export default class VerifiableCredentialsService implements IBackupable {
     });
   };
 
+  generateVerifiablePresentationRequest = async (
+    verifiablePresentationRequest: IVerifiablePresentationRequest,
+  ): Promise<void> => {
+    await this.browserController.openPopup({
+      params: {
+        redirect: Paths.GENERATE_VERIFIABLE_PRESENTATION_REQUEST,
+        request: verifiablePresentationRequest.request,
+      },
+    });
+  };
+
+  generateVerifiablePresentation = async (verifiablePresentation: IVerifiablePresentation): Promise<void> => {
+    await this.historyService.trackOperation(OperationType.GENERATE_VERIFIABLE_PRESENTATION, {});
+    await this.notificationService.create({
+      options: {
+        title: "Verifiable Presentation generated.",
+        message: `Generated 1 Verifiable Presentation.`,
+        iconUrl: browser.runtime.getURL("/icons/logo.png"),
+        type: "basic",
+      },
+    });
+    const tabs = await browser.tabs.query({ active: true });
+    await Promise.all(
+      tabs.map((tab) =>
+        browser.tabs
+          .sendMessage(tab.id!, {
+            type: EventName.GENERATE_VERIFIABLE_PRESENTATION,
+            payload: { verifiablePresentation },
+          })
+          .catch(() => undefined),
+      ),
+    );
+  };
```

Comment:
Why this method has only user notifications? Is it expected that we don't save anything here?

Reply:
Yes. The generated Verifiable Presentation is just emitted as an event to be consumer by the verifier. The extension doesn't need to save anything, except for the history log.